File: test/aidbox_sdk/snapshots/typescripthl7-fhir-r4-core/Provenance.ts
import { Period } from "./Period";
import { CodeableConcept } from "./CodeableConcept";
import { Signature } from "./Signature";
import { DomainResource } from "./DomainResource";
import { Reference } from "./Reference";
import { BackboneElement } from "./BackboneElement";

export type ProvenanceAgent = BackboneElement & {
    who: Reference;
    role?: CodeableConcept[];
    type?: CodeableConcept;
    onBehalfOf?: Reference;
};

export type ProvenanceEntity = BackboneElement & {
    role: string;
    what: Reference;
    agent?: Reference[];
};

export type Provenance = DomainResource & {
    occurred?: string | Period;
    signature?: Signature[];
    recorded: string;
    agent: ProvenanceAgent[];
    policy?: string[];
    reason?: CodeableConcept[];
    activity?: CodeableConcept;
    target: Reference[];
    location?: Reference;
    entity?: ProvenanceEntity[];
};

File: test/aidbox_sdk/snapshots/python/hl7_fhir_r4_core/Provenance.py
from typing import Optional, List
from dataclasses import dataclass, field
from base import Period
from base import CodeableConcept
from base import Signature
from base import DomainResource
from base import Reference
from base import BackboneElement

@dataclass(kw_only=True)
class ProvenanceAgent(BackboneElement):
    who: Reference
    role: Optional[List[CodeableConcept]] = field(default_factory=list)
    type: Optional[CodeableConcept] = None
    on_behalf_of: Optional[Reference] = None

@dataclass(kw_only=True)
class ProvenanceEntity(BackboneElement):
    role: str
    what: Reference
    agent: Optional[List[Reference]] = field(default_factory=list)

@dataclass(kw_only=True)
class Provenance(DomainResource):
    signature: Optional[List[Signature]] = field(default_factory=list)
    occurred_date_time: Optional[str] = None
    recorded: str
    agent: list[ProvenanceAgent] = field(default_factory=list)
    policy: Optional[List[str]] = field(default_factory=list)
    reason: Optional[List[CodeableConcept]] = field(default_factory=list)
    activity: Optional[CodeableConcept] = None
    target: list[Reference] = field(default_factory=list)
    location: Optional[Reference] = None
    entity: Optional[List[ProvenanceEntity]] = field(default_factory=list)
    occurred_period: Optional[Period] = None